src/runtime/CollectionWrappers/SequenceWrapper.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Collections.Generic;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class SequenceWrapper<T> : IterableWrapper<T>, ICollection<T>
+    {
+        public SequenceWrapper(PyObject pyObj) : base(pyObj)
+        {
+
+        }
+
+        public int Count
+        {
+            get
+            {
+                var size = Runtime.PySequence_Size(pyObject.Handle);
+                if (size == -1)
+                {
+                    Runtime.CheckExceptionOccurred();
+                    throw new Exception("Unable to get sequence size!");
+                }
+
+                return (int)size;
+            }
+        }
+
+        public virtual bool IsReadOnly => false;
+
+        public virtual void Add(T item)
+        {
+            //not implemented for Python sequence.
+            //ICollection is the closest analogue but it doesn't map perfectly.
+            //SequenceWrapper is not final and can be subclassed if necessary
+            throw new NotImplementedException();
+        }
+
+        public void Clear()
+        {
+            if (IsReadOnly)
+                throw new NotImplementedException();
+            var result = Runtime.PySequence_DelSlice(pyObject.Handle, 0, Count);
+            if (result == -1)
+            {
+                Runtime.CheckExceptionOccurred();
+                throw new Exception("failed to clear sequence");
+            }
+        }
+
+        public bool Contains(T item)
+        {
+            //not sure if IEquatable is implemented and this will work!
+            foreach (var element in this)
+                if (element.Equals(item)) return true;
+
+            return false;
+        }
+
+        private T getItem(int index)
+        {
+            IntPtr item = Runtime.PySequence_GetItem(pyObject.Handle, index);
+            object obj;
+
+            if (!Converter.ToManaged(item, typeof(T), out obj, true))
+            {
+                Runtime.XDecref(item);
+                Runtime.CheckExceptionOccurred();
+            }
+
+            return (T)obj;
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            for (int index = 0; index < Count; index++)
+            {
+                array[index + arrayIndex] = getItem(index);
+            }
+        }
+
+        protected bool removeAt(int index)
+        {
+            if (IsReadOnly)
+                throw new NotImplementedException();
+            if (index >= Count || index < 0)
+                throw new IndexOutOfRangeException();
+
+            return Runtime.PySequence_DelItem(pyObject.Handle, index) != 0;
```

Comment:
Failure of `PySequence_DelItem` is an exception too.

Reply:
@lostmsu in that case I will clear the exception in Remove (In C# it is idiomatic to return false in Remove rather than to throw) but throw it in ListWrapper RemoveAt

Reply:
That depends on the reason for the failure. It should not throw only if the item was not in the collection in the first place. If it was in the collection, but something prevented it from being removed, it must throw.